src/mlpack/methods/ann/visitor/gradient_visitor.hpp

```diff
@@ -30,7 +30,7 @@ class GradientVisitor : public boost::static_visitor<void>
  public:
   //! Executes the Gradient() method of the given module using the input and
   //! delta parameter.
-  GradientVisitor(arma::mat&& input, arma::mat&& delta);
+  GradientVisitor(arma::mat&& input, arma::mat&& delta, int layer = -1);
```

Comment:
Do you think instead of using ``int layer = -1`` we could provide a second constructor:

```c++
GradientVisitor(arma::mat&& input, arma::mat&& delta) : hasIndex(false), index(0)
{
}

GradientVisitor(arma::mat&& input, arma::mat&& delta, const size_t index) : hasIndex(true), index(index)
{
}
```

Let me know what you think.

Reply:
alright sounds good to me

Reply:
@zoq i have made the changes